pac4j-oauth/src/main/java/org/pac4j/oauth/client/FacebookClient.java

```diff
@@ -81,6 +83,8 @@
     protected boolean requiresExtendedToken = false;
     
     protected StateApi20 api20;
+
+    protected String appsecretProof = null;
```

Comment:
Facebook defines three different bits of data: app ID, app secret, and client token. I am not conversant with the OAUTH protocol nor with the internals of pac4j. I am only just able to understand some of how Shiro works. None of the documentation is at all clear to one such as me; it may be plenty useful to someone who already has the whole model of OAUTH, pac4j, and Shiro loaded mentally and just needs a reminder but it is frustratingly vague to one for whom those systems are tangential to the work at hand.

That said, I looked at the code at line 145:
            url = CommonHelper.addParameter(url, OAuthConstants.CLIENT_SECRET, this.secret);
and since that was already in place I did not assume that it was incorrect. Is it, in fact, incorrect? OAuthConstants.CLIENT_SECRET is the string "client_secret" and if that parameter is not actually used by the Facebook OAUTH protocol, then the most straightforward change would be to add a new constant, OAuthConstants.FB_APPSECRET_PROOF for the "appsecret_proof" parameter and use that instead. I am unaware of whether any other OAUTH services use a parameter of that name; if not, it may make sense to define the constant within FacebookClient so as to reduce confusion, having a Facebook-only parameter sitting in a global table of OAUTH constants.

Are you aware of the answers to any or all of these questions?